msamp/deepspeed/runtime/zero/fp8_stage_1_and_2.py

```diff
@@ -462,33 +463,17 @@ def fp8_average_tensor(self, tensor):
 
             tensor_to_reduce = tensor
 
-            # MS-AMP does not support distributed operators with process group
-            # fallback to FP16 communication
-            meta = ScalingMeta(Dtypes.kfloat8_e4m3)
-
-            def _fp8_to_fp16(tensor):
-                return TransformerEngineWrapper.cast_from_fp8(
-                    tensor.view(1, -1), meta.scale_inv, meta.qtype, Dtypes.kfloat16
-                ).view_as(tensor)
-
-            def _fp16_to_fp8(tensor):
-                return TransformerEngineWrapper.cast_to_fp8(
-                    tensor.view(1, -1), meta.scale, meta.amax[0], meta.scale_inv, meta.qtype
-                ).view_as(tensor)
-
+            # Reduce gradient.
             async_handles = []
-            grad_slice_pairs = []
             for i, (dst, bucket_offset, numel) in enumerate(rank_and_offsets):
                 grad_slice = tensor_to_reduce.narrow(0, int(bucket_offset), int(numel))
                 dst_rank = dist.get_global_rank(real_dp_process_group[i], dst)
-                fp16_grad_slice = _fp8_to_fp16(grad_slice)
-                async_handle = dist.reduce(fp16_grad_slice, dst=dst_rank, group=real_dp_process_group[i], async_op=True)
+                async_handle = DistOp.reduce(
```

Comment:
please replace `Dtypes.kfloat8_e4m3` with `WEIGHT_GRAD_QTYPE`.
It is suitable to change the data type of weight gradient by a variable.

Reply:
Good suggestion. Fixed.